### ISSUE TITLE
Creating a required OneToMany in the "Many" part.

### ISSUE THREAD

**Issue by marcosmoreno** [jhipster/jhipster-domain-language#8]
If I create the example:

```
entity Owner {}
entity Car {}
relationship OneToMany {
  Owner{car} to Car{owner}
}
```

Jhipster creates Entities for Owner and Car, and I can create Cars without owner. 
Is it possible to specify that the owner is required with JDL?

In the Entities annotations, I see that a ManyToOne is created for the Car Entity. I guess I would like that this relationship has (optional=false)  next to the declaration, and that Jhipster automatically cares of the front-end code validation.

**Comment by MathieuAA** [comment]
Hello, you're not the first one who asked for this. We could add this for JHipster UML and the JDL if JHipster implements it. 

Le 14 avril 2016 22:06:05 GMT+02:00, marcosmoreno <notifications@github.com> a écrit :
>If I create the example:
>
>```
>entity Owner {}
>entity Car {}
>relationship OneToMany {
>  Owner{car} to Car{owner}
>}
>```
>
>Jhipster creates Entities for Owner and Car, and I can create Cars
>without owner. 
>Is it possible to specify that the owner is required with JDL?
>
>In the Entities annotations, I see that a ManyToOne is created for the
>Car Entity. I guess I would like that this relationship has
>(optional=false)  next to the declaration, and that Jhipster
>automatically cares of the front-end code validation.
>
>---
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/jhipster/jhipster-domain-language/issues/8

-- 
Mathieu ABOU-AICHI

**Comment by marcosmoreno** [comment]
Thank you @MathieuAA . I created the feature request in the generator-jhipster.

Just some info: when I just add the annotation @NotNull to the @ManyToOne field and try to add a registry to the Car entity without selecting the Owner, jhipster behaves exactly as I wanted, as it tells that the field cannot be empty (jhipster 3.0). 

This means that jhipster is already taking care of it, at least the bit of inserting with the entities front end. Is then just missing that the jhipster:import-jdl includes the capability of adding the @NotNull to the @ManyToOne annotation when "something" is parsed in the JDL? Or might be some unexpected consequences somewhere doing this?

**Comment by marcosmoreno** [comment]
@MathieuAA  would the fix in the referenced generator issue suffice for adding this to the Jhipster UML? Like adding <validation> to the JDL in the relationships declaration?

**Comment by deepu105** [comment]
The support is added in JHipster but we have to do some work to make it
work in JDL so it will take some more time based on our availability to
work on this.

Thanks & Regards,
Deepu

**Comment by MathieuAA** [comment]
It's easily done... I'll see about doing it today or tomorrow.

**Comment by MathieuAA** [comment]
Closed as it has been added (see #22).
